packages/resin-event-log/test/resin-event-log.spec.js

```diff
@@ -0,0 +1,243 @@
+var _ = require('lodash')
+var expect = require('chai').expect
+var base64Decode = require('base-64').decode
+var querystring = require('querystring')
+
+var mock = require('resin-universal-http-mock')
+
+var IS_BROWSER = typeof window !== 'undefined'
+
+if (IS_BROWSER) {
+	window.MIXPANEL_CUSTOM_LIB_URL = 'http://cdn.mxpnl.com/libs/mixpanel-2-latest.js'
+	// window.GA_CUSTOM_LIB_URL = 'https://www.google-analytics.com/analytics_debug.js'
+}
+
+var ResinEventLog = require('..')
+
+var MIXPANEL_TOKEN = 'MIXPANEL_TOKEN'
+var SYSTEM = 'TEST'
+var MIXPANEL_HOST = 'http://api.mixpanel.com'
+var GA_ID = 'UA-123456-0'
+var GA_SITE = 'resintest.io'
+var GA_HOST = 'https://www.google-analytics.com'
+var FAKE_USER = {
+	username: 'fake',
+	id: 123,
+	email: 'fake@example.com',
+	$created: new Date().toISOString()
+}
+
+function validateMixpanelQuery(event) {
+	return function(queryObject) {
+		var data = queryObject.data
+		if (!data) return false
+
+		try {
+			data = JSON.parse(base64Decode(data))
+			return (
+				data && data.properties &&
+				data.properties.token === MIXPANEL_TOKEN &&
+				(!event || event === data.event)
+			)
+		} catch (e) {
+			return false
+		}
+	}
+}
+
+function createMixpanelMock(options) {
+	_.defaults(options, {
+		host: MIXPANEL_HOST,
+		method: 'GET',
+		filterQuery: validateMixpanelQuery(options.event),
+		response: '1'
+	})
+	delete options.event
+	return mock.create(options)
+}
+
+function validateGaBody(bodyString) {
+	var data = bodyString.split('\n')[0]
+	if (!data) return false
+
+	try {
+		data = querystring.parse(data)
+
+		return (
+			data &&
+			data.t === 'event' &&
+			data.tid === GA_ID &&
+			data.ec === GA_SITE &&
+			data.el === SYSTEM
+		)
+	} catch (e) {
+		return false
+	}
+}
+
+function createOneGaMock(endpoint) {
+	return mock.create({
+		host: GA_HOST,
+		endpoint: endpoint,
+		method: 'POST',
+		filterBody: validateGaBody
+	})
+}
+
+function createGaMock(endpoint) {
+	var mocks = [
+		createOneGaMock(endpoint),
+		createOneGaMock('/r' + endpoint)
+	]
+	return {
+		isDone: function() {
+			return _.some(mocks, function(mock) {
+				return mock.isDone()
+			})
+		}
+	}
+}
+
+describe('ResinEventLog', function () {
+	// Use 0 for debugging in the real browser
+	this.timeout(3000)
+
+	before(function() {
+		mock.init()
+	})
+	afterEach(function() {
+		mock.reset()
+	})
+	after(function() {
+		mock.teardown()
+	})
```

Comment:
Yeah. Puts an extra assumption these methods are static (not dependent on the context), but it's safe in this case as `mock` is singleton